tests/testsuite/namespace.go

```diff
@@ -200,12 +201,23 @@ func removeNamespaces() {
 
 	// Wait until the namespaces are terminated
 	fmt.Println("")
-	for _, namespace := range TestNamespaces {
-		fmt.Printf("Waiting for namespace %s to be removed, this can take a while ...\n", namespace)
+	for _, namespaceName := range TestNamespaces {
+		fmt.Printf("Waiting for namespace %s to be removed, this can take a while ...\n", namespaceName)
 		EventuallyWithOffset(1, func() error {
-			return virtCli.CoreV1().Namespaces().Delete(context.Background(), namespace, metav1.DeleteOptions{})
-		}, 240*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), fmt.Sprintf("should successfully delete namespace '%s'", namespace))
+			return virtCli.CoreV1().Namespaces().Delete(context.Background(), namespaceName, metav1.DeleteOptions{})
+		}, 240*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), fmt.Sprintf("should successfully delete namespace '%s': \n%s", namespaceName, dumpNamespaceObject(namespaceName)))
+	}
+}
+
+func dumpNamespaceObject(namespaceName string) string {
```

Comment:
we might want to also add the following code:
Check if the namespace has finalizers, and if so, print all the resources with this finalizer
for example on kubvevirtci when a namespace failed to be deleted, this what helped me manually the find the cause

Reply:
please see the example of the namespace object print, you can see that it already points to stuff like finalizers stuck etc..

Reply:
I saw, just in some cases it helps to know which crds for example have those finalizers,
anyhow lets do incremental